specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json

```diff
@@ -939,37 +932,43 @@
           }
         },
         "x-ms-examples": {
-          "Get app settings of a static site build": {
-            "$ref": "./examples/ListStaticSiteBuildAppSettings.json"
+          "Get overview of database connections for the static site build.": {
+            "$ref": "./examples/GetStaticSiteBuildDatabaseConnection.json"
           }
         }
-      }
-    },
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/staticSites/{name}/builds/{environmentName}/listFunctionAppSettings": {
-      "post": {
+      },
+      "put": {
         "tags": [
           "StaticSites"
         ],
-        "summary": "Gets the application settings of a static site build.",
-        "description": "Description for Gets the application settings of a static site build.",
-        "operationId": "StaticSites_ListStaticSiteBuildFunctionAppSettings",
+        "summary": "Create or update a database connection for a static site build",
+        "description": "Description for Create or update a database connection for a static site build",
+        "operationId": "StaticSites_CreateOrUpdateBuildDatabaseConnection",
         "parameters": [
           {
             "$ref": "#/parameters/resourceGroupNameParameter"
           },
           {
             "name": "name",
             "in": "path",
-            "description": "Name of the static site.",
+            "description": "Name of the static site",
             "required": true,
             "type": "string"
           },
           {
-            "name": "environmentName",
-            "in": "path",
-            "description": "The stage site identifier.",
+            "$ref": "#/parameters/environmentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/databaseConnectionNameParameter"
+          },
+          {
+            "name": "databaseConnectionRequestEnvelope",
+            "in": "body",
+            "description": "A JSON representation of the database connection request properties",
             "required": true,
-            "type": "string"
+            "schema": {
+              "$ref": "#/definitions/DatabaseConnectionRequest"
+            }
```

Comment:
>} [](http://example.com/codeflow?start=12&length=1)

Why do you need a separate object for request and response ? Ideally you should be having the same object in REST that is PUT and GET except for any secrets or readonly properties